client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -311,16 +362,18 @@ class DeltaSharingRestClient(
       maxFiles = None,
       pageToken = None,
       includeRefreshToken = Some(includeRefreshToken),
-      refreshToken = refreshToken
+      refreshToken = refreshToken,
+      idempotencyKey = idempotencyKey,
+      requestedColumns = Seq.empty[String]
     )
     val (version, respondedFormat, lines, refreshTokenOpt) = if (queryTablePaginationEnabled) {
       logInfo(
         s"Making paginated queryTable requests for table " +
         s"${table.share}.${table.schema}.${table.name} with maxFiles=$maxFilesPerReq"
       )
-      getFilesByPage(target, request)
+      getFilesByPage(table, target, request)
     } else {
-      val (version, respondedFormat, lines) = getNDJson(target, request)
+      val (version, respondedFormat, lines, _) = getNDJsonWithAsync(table, target, request)
```

Comment:
Will this affect the existing queries?

Reply:
this is designed to be compatible with existing sync query path, and existing test cases succeed.

Reply:
I think it does make the code a bit risky; if there is a bug in getNDJsonWithAsync, it will impact existing queries.
Should we do an if (enableAsync) switch here instead?

Reply:
I agree that this seems a bit risky to do